app/models/storage.rb

```diff
@@ -525,38 +535,38 @@ def smartstate_analysis(miq_task_id = nil)
       miq_task.state_active unless miq_task.nil?
     end
 
-    hosts = active_hosts_with_authentication_status_ok_in_zone(MiqServer.my_zone)
-    if hosts.empty?
-      message = "There are no active Hosts with valid credentials connected to Storage: [#{name}] in Zone: [#{MiqServer.my_zone}]."
+    emss = ext_management_systems_with_authentication_status_ok_in_zone(MiqServer.my_zone)
+    if emss.empty?
+      message = "There are no EMSs with valid credentials connected to Storage: [#{name}] in Zone: [#{MiqServer.my_zone}]."
       _log.warn(message)
       raise MiqException::MiqUnreachableStorage,
-            _("There are no active Hosts with valid credentials connected to Storage: [%{name}] in Zone: [%{zone}].") %
+            _("There are no EMSs with valid credentials connected to Storage: [%{name}] in Zone: [%{zone}].") %
               {:name => name, :zone => MiqServer.my_zone}
     end
 
-    max_parallel_storage_scans_per_host = self.class.max_parallel_storage_scans_per_host
-    host = nil
-    hosts.each do |h|
-      next if smartstate_analysis_count_for_host_id(h.id) >= max_parallel_storage_scans_per_host
-      host = h
+    max_parallel_storage_scans_per_ems = self.class.max_parallel_storage_scans_per_ems
+    ems = nil
+    emss.each do |e|
+      next if smartstate_analysis_count_for_ems_id(e.id) >= max_parallel_storage_scans_per_ems
+      ems = e
       break
     end
 
-    if host.nil?
+    if ems.nil?
       raise MiqException::MiqQueueRetryLater.new(:deliver_on => Time.now.utc + 1.minute) if qmessage?(method_name)
-      host = hosts.random_element
+      ems = emss.random_element
     end
 
-    $_miq_worker_current_msg.update_attributes!(:target_id => host.id) if qmessage?(method_name)
+    $_miq_worker_current_msg.update_attributes!(:target_id => ems.id) if qmessage?(method_name)
 
     st = Time.now
-    message = "Storage [#{name}] via Host [#{host.name}]"
+    message = "Storage [#{name}] via EMS [#{ems.name}]"
     _log.info("#{message}...Starting")
-    scan_starting(miq_task_id, host)
-    if host.respond_to?(:refresh_files_on_datastore)
-      host.refresh_files_on_datastore(self)
+    scan_starting(miq_task_id, ems)
+    if ems.respond_to?(:refresh_files_on_datastore)
+      ems.refresh_files_on_datastore(self)
```

Comment:
Is VmWare the only provider that needs to worry about this?  Any chance that this will mess up, say, RHV?

Reply:
I checked the rest of the providers and VMware is the only one which implements this method.